runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/windmill/grpcclient/GrpcWindmillServerTest.java

```diff
@@ -136,6 +146,65 @@ private <Stream extends StreamObserver> void maybeInjectError(Stream stream) {
     }
   }
 
+  @Test
+  public void testWindmillApplianceGrpcDeadlineInterceptor() {
+    Windmill.GetWorkResponse getWorkResponse =
+        Windmill.GetWorkResponse.newBuilder()
+            .addWork(
+                Windmill.ComputationWorkItems.newBuilder()
+                    .setComputationId("someComputation1")
+                    .build())
+            .build();
+
+    serviceRegistry.addService(
+        new WindmillApplianceGrpc.WindmillApplianceImplBase() {
+          @Override
+          public void getWork(
+              Windmill.GetWorkRequest request,
+              StreamObserver<Windmill.GetWorkResponse> responseObserver) {
+            responseObserver.onNext(getWorkResponse);
+            responseObserver.onCompleted();
+          }
+        });
+
+    ClientInterceptor testInterceptor =
+        new ClientInterceptor() {
+          Deadline deadline;
+
+          @Override
+          public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(
+              MethodDescriptor<ReqT, RespT> methodDescriptor,
+              CallOptions callOptions,
+              Channel channel) {
+            if (deadline != null) {
+              // We want to assert that a new deadline is being created on every call since
+              // deadlines are absolute.
+              assertNotEquals(deadline, callOptions.getDeadline());
+            }
+
+            deadline = callOptions.getDeadline();
```

Comment:
nit: verify callOptions.getDeadline is non-null

Reply:
done thanks Sam!